chain/chain/src/types.rs

```diff
@@ -454,18 +453,14 @@ pub struct ShardStateSyncResponseHeader {
     pub prev_chunk_proof: MerklePath,
     pub incoming_receipts_proofs: Vec<ReceiptProofResponse>,
     pub root_proofs: Vec<Vec<RootProof>>,
-}
-
-#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
-pub struct ShardStateSyncResponsePart {
-    pub state_part: StatePart,
-    pub proof: Vec<u8>,
+    pub state_root_node: StateRootNode,
```

Comment:
Somewhere in the code we have logic to validate all of these proofs from a received header.
That logic needs to be updated to also validate that `StateRootNode` data matches its hash and extracted `memory_usage`

Reply:
Discussed offline. I will add `validate_state_root_node` method to runtime adapter.